src/core/ext/transport/chttp2/server/chttp2_server.cc

```diff
@@ -868,10 +868,10 @@ grpc_error_handle Chttp2ServerAddPort(Server* server, const char* addr,
     grpc_error_handle error = GRPC_ERROR_NONE;
     if (absl::StartsWith(parsed_addr, kUnixUriPrefix)) {
       error = grpc_resolve_unix_domain_address(
-          addr + sizeof(kUnixUriPrefix) - 1, &resolved);
+          &parsed_addr[sizeof(kUnixUriPrefix) - 1], &resolved);
```

Comment:
I suggest writing this as `parsed_addr.data() + sizeof(kUnixUriPrefix) - 1`.

Reply:
Actually, both of these solutions break when `addr` contains embedded null. Passing a c-str is not a viable solution here.

Reply:
`absl::ConsumePrefix` does the job more efficiently than the previous iteration, only checking `absl::StartsWith(...)` once.